nemo_skills/inference/prompt/few_shot_examples/examples_tabmwp.py

```diff
@@ -0,0 +1,217 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+text_with_code = [
+    {
+        "question": """Read the table regarding "Coin collections" and then write Python code to answer the following question:
+
+Question: Some friends discussed the sizes of their coin collections. What is the mean of the numbers?""",
+        "table": """
+Name | Number of coins
+Braden | 76
+Camilla | 94
+Rick | 86
+Mary | 84
+Hector | 80
+Devin | 83
+Emily | 82
+Avery | 87
+""".strip(),
+        "reference_solution": "Read the numbers from the table.\n\n76, 94, 86, 84, 80, 83, 82, 87\n\nFirst, count how many numbers are in the group.\n\nThere are 8 numbers.\n\nNow add all the numbers together:\n\n76 + 94 + 86 + 84 + 80 + 83 + 82 + 87 = 672\n\nNow divide the sum by the number of numbers:\n\n672 \u00f7 8 = 84\n\nThe mean is 84.",
+        "generated_solution": """
+Let's solve this problem using Python code and the values from the table.
+{start_code}
+number_of_coins_for_different_person = [76, 94, 86, 84, 80, 83, 82, 87]
+result = sum(number_of_coins_for_different_person) / len(number_of_coins_for_different_person)
+result
+{end_code}
+{start_code_output}
+84
+{end_code_output}
+So the mean of the sizes of their coin collections is \\boxed{84} coins.
+""".strip(),
+    },
+    {
+        "question": """Read the table regarding "" and then write Python code to answer the following question:
+
+Question: Look at the table. Then answer the question. At a price of $155, is there a shortage or a surplus? Choose from the the options: [shortage, surplus]""",
+        "table": """
+Price | Quantity demanded | Quantity supplied
+$155 | 22,600 | 5,800
+$275 | 20,500 | 9,400
+$395 | 18,400 | 13,000
+$515 | 16,300 | 16,600
+$635 | 14,200 | 20,200
+""".strip(),
+        "reference_solution": "At the price of $155, the quantity demanded is greater than the quantity supplied. There is not enough of the good or service for sale at that price. So, there is a shortage.\n",
+        "generated_solution": """
+Let's solve this problem using Python code and the values from the table.
+{start_code}
+quantity_demanded_price_155 = 22600
```

Comment:
Note that you don't have to use code if the problem does not require it. In this case it looks like a more natural solution would be to just solve it in text

Reply:
Do we have a way of mixing code/text solutions with the model being able to choose what to use?

In this case the end goal is to have the values loaded programmatically so I figure these examples should all involve translating the problem to code to begin with.

Reply:
The model is supposed to learn that. So if you generate some examples that use code and some that don't and train on those, the hope is that it will be able to differentiate between these two cases at inference time. But you need to be showing this in few shots (either in the same set of few shots show both or have example with/without code and run multiple labeling jobs)

Reply:
Gotcha, makes sense. I'll keep these with code examples for now since we're focusing on code-generation, but good to keep in mind.